mpy-cross/main.c

```diff
@@ -350,6 +364,29 @@ MP_NOINLINE int main_(int argc, char **argv) {
         }
     }
 
+    if (arch_flags && mp_dynamic_compiler.native_arch != MP_NATIVE_ARCH_NONE) {
+        bool processed = false;
+        #if MICROPY_EMIT_NATIVE && MICROPY_EMIT_RV32
+        if (mp_dynamic_compiler.native_arch == MP_NATIVE_ARCH_RV32IMC) {
+            mp_dynamic_compiler.backend_options = (void *)&rv32_options;
+            if (strncmp(arch_flags, "zba", sizeof("zba") - 1) == 0) {
+                rv32_options.emit_zba = true;
+                processed = true;
+            }
+        }
+        #endif
+        if (!processed) {
+            mp_printf(&mp_stderr_print, "unrecognised arch flags\n");
+            exit(1);
+        }
+        mp_printf(&mp_stderr_print,
+            "Warning: Using architecture-specific flags may create a MPY file whose code\n"
+            "         won't run on all target boards!\n"
```

Comment:
"targets" instead of "target boards"

Reply:
This should have been addressed in the latest version of the PR.